# Phenotype exploratory analysis

### Exploring list of patients from phenotype classification task from Gehrmann et al. (2018)

Downloaded as 'annotations.csv' from https://github.com/sebastianGehrmann/phenotyping/blob/master/data/annotations.csv

In [1]:
import pandas as pd
import numpy as np
import os
import psycopg2
from sqlalchemy import create_engine 
import string
import spacy
import re
from datetime import date, datetime, timedelta
import random

In [54]:
annotations = pd.read_csv('../data/annotations.csv')
annotations.columns.values[0] = 'hadm_id'
annotations.columns.values[1] = 'subject_id'
annotations.head()

hadm_id  subject_id  chart.time  cohort  Obesity  Non.Adherence  \
0   118003        3644      118003       1        0              0   
1   177830       97736      999999       0        0              0   
2   185673       27694      999999       0        0              0   
3   131938       16275      131938       0        0              0   
4   198999        4059      198999       1        0              0   

   Developmental.Delay.Retardation  Advanced.Heart.Disease  \
0                                0                       0   
1                                0                       0   
2                                0                       0   
3                                0                       0   
4                                0                       0   

   Advanced.Lung.Disease  Schizophrenia.and.other.Psychiatric.Disorders  \
0                      0                                              0   
1                      0                                              0   
2                      0                                              0   
3                      0                                              0   
4                      0                                              1   

   Alcohol.Abuse  Other.Substance.Abuse  Chronic.Pain.Fibromyalgia  \
0              0                      0                          1   
1              0                      0                          0   
2              0                      0                          0   
3              0                      0                          0   
4              0                      0                          0   

   Chronic.Neurological.Dystrophies  Advanced.Cancer  Depression  Dementia  \
0                                 0                0           1         0   
1                                 0                1           0         0   
2                                 0                0           1         0   
3                                 0                0           0         0   
4                                 0                0           1         0   

   Unsure  
0       0  
1       0  
2       0  
3       0  
4       0

**Cohort 1** refers to 'frequent flyers' and **Cohort 0** refers to random summaries from patients who are not frequent flyers

In [43]:
subjects = list(annotations['subject.id'].drop_duplicates())
subject_str = '(' + ", ".join(str(x) for x in subjects) + ')'
len(subjects)

1045

Converting list of unique subject ids to a string. Only 1045 unique patients from 1610 total

In [11]:
# connect to the mimic database and set the search path to the 'mimiciii' schema

dbschema='mimiciii'
cnx = create_engine('postgresql+psycopg2://aa5118:mimic@localhost:5432/mimic',
                    connect_args={'options': '-csearch_path={}'.format(dbschema)})

Confirming that it is only adults who are part of the phenotype classification task

In [44]:
df = pd.read_sql_query('''
    SELECT p.subject_id, p.dob, p.gender,
           ROUND((cast(chartdate as date) - cast(dob as date)) / 365.242,0) AS age_at_noteevent
    FROM patients p 
    INNER JOIN noteevents n 
    ON p.subject_id = n.subject_id 
    WHERE p.subject_id IN ''' + subject_str + '''
    ORDER BY ROUND((cast(chartdate as date) - cast(dob as date)) / 365.242,0)
''', cnx)
df.head()

subject_id        dob gender  age_at_noteevent
0       10510 2183-07-09      M              18.0
1       10510 2183-07-09      M              18.0
2       10510 2183-07-09      M              18.0
3       10510 2183-07-09      M              18.0
4       10510 2183-07-09      M              18.0

###### Now, we compare the manually annotated phenotypes with the diagnoses from the MIMIC table

In [58]:
df_diag = pd.read_sql_query('''
    SELECT * 
    FROM diagnoses_icd 
    JOIN d_icd_diagnoses 
    USING (icd9_code)
    ORDER BY subject_id, hadm_id
    ''', cnx)
df_diag.head()

icd9_code  row_id  subject_id  hadm_id  seq_num  row_id  \
0     V3001       1           2   163353        1   13695   
1      V053       2           2   163353        2   12202   
2      V290       3           2   163353        3   13688   
3      0389       4           3   145834        1     660   
4     78559       5           3   145834        2   12992   

                short_title                                         long_title  
0    Single lb in-hosp w cs  Single liveborn, born in hospital, delivered b...  
1  Need prphyl vc vrl hepat  Need for prophylactic vaccination and inoculat...  
2    NB obsrv suspct infect     Observation for suspected infectious condition  
3            Septicemia NOS                             Unspecified septicemia  
4      Shock w/o trauma NEC              Other shock without mention of trauma

In [56]:
annotations_merged = pd.merge(annotations, df_diag, on=['subject_id', 'hadm_id'])

In [62]:
pd.options.display.max_columns = None
annotations_merged[annotations_merged['subject_id'] == 97736]

hadm_id  subject_id  chart.time  cohort  Obesity  Non.Adherence  \
9    177830       97736      999999       0        0              0   
10   177830       97736      999999       0        0              0   
11   177830       97736      999999       0        0              0   
12   177830       97736      999999       0        0              0   
13   177830       97736      999999       0        0              0   
14   177830       97736      999999       0        0              0   
15   177830       97736      999999       0        0              0   
16   177830       97736      999999       0        0              0   
17   177830       97736      999999       0        0              0   
18   177830       97736      999999       0        0              0   
19   177830       97736      999999       0        0              0   
20   177830       97736      999999       0        0              0   

    Developmental.Delay.Retardation  Advanced.Heart.Disease  \
9                                 0                       0   
10                                0                       0   
11                                0                       0   
12                                0                       0   
13                                0                       0   
14                                0                       0   
15                                0                       0   
16                                0                       0   
17                                0                       0   
18                                0                       0   
19                                0                       0   
20                                0                       0   

    Advanced.Lung.Disease  Schizophrenia.and.other.Psychiatric.Disorders  \
9                       0                                              0   
10                      0                                              0   
11                      0                                              0   
12                      0                                              0   
13                      0                                              0   
14                      0                                              0   
15                      0                                              0   
16                      0                                              0   
17                      0                                              0   
18                      0                                              0   
19                      0                                              0   
20                      0                                              0   

    Alcohol.Abuse  Other.Substance.Abuse  Chronic.Pain.Fibromyalgia  \
9               0                      0                          0   
10              0                      0                          0   
11              0                      0                          0   
12              0                      0                          0   
13              0                      0                          0   
14              0                      0                          0   
15              0                      0                          0   
16              0                      0                          0   
17              0                      0                          0   
18              0                      0                          0   
19              0                      0                          0   
20              0                      0                          0   

    Chronic.Neurological.Dystrophies  Advanced.Cancer  Depression  Dementia  \
9                                  0                1           0         0   
10                                 0                1           0         0   
11                                 0                1           0         0   
12                             

According to the manual annotation, the patient has advanced cancer. This corresponds to the second most severe diagnosis from the mimic diagnoses table - 'secondary malignant neoplasm' etc.

The most severe diagnosis appears to be 'Intestinal infection due to Clostridium difficile' which does not appear in the manual annotations. The reverse of this is also true for other patients. This indicates that perhaps we should add the diagnoses of the patients to the input of the text generation model